shotover-proxy/src/transforms/chain.rs

```diff
@@ -154,6 +154,20 @@ impl TransformChain {
                 debug!(
                     "buffered chain processing thread exiting, stopping chain loop and dropping"
                 );
+
```

Comment:
We need to document now that transforms can get called with an empty list of messages on shutdown and flush set. If they don't expect it or handle it could lead to delayed shutdowns or bugs.

Reply:
Added a line to the Transform::transform docs